src/persist/src/indexed/mod.rs

```diff
@@ -171,9 +176,35 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
     /// Attempt to commit the current in-memory metadata state to durable storage,
     /// and if not, revert back to a previous version.
     fn try_set_meta(&mut self, prev: BlobMeta) -> Result<(), Error> {
+        let meta = self.serialize_meta();
+
+        self.metrics
+            .stream_count
+            .set(u64::cast_from(meta.id_mapping.len()));
+        let future_blob_count: usize = meta.futures.iter().map(|x| x.batches.len()).sum();
+        self.metrics
+            .future_blob_count
+            .set(u64::cast_from(future_blob_count));
+        let future_blob_bytes: u64 = meta
```

Comment:
this metric (edit: and also a few more like stream count, future_blob_count) look they're never updated if im reading this all correctly. I think its fine to merge as is, but a todo would be nice as I wasn't sure if the intent was to have a "on startup this was the value" metric as well or if this was just wip

Reply:
hmmm, I don't think I follow? they get updated every time try_set_meta is called

Reply:
ah yes indeed this was a brain fart on my part - i thought we were in `new` -- whoops!

Reply:
but you probably want to update the metrics after you've committed the metadata right? just in case you had to restore?

Reply:
I did think of that and decided it's probably more interesting to keep track of the reqs, like in case we try to write some huge meta and it keeps failing because it's too big. Probably worth a comment though

Reply:
scratch that. tried to write the comment and decided you were right. done